src/test/java/MinesweeperControllerTest.java

```diff
@@ -3,18 +3,32 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.testfx.framework.junit5.ApplicationExtension;
 import java.util.ArrayList;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.mockito.Mockito.mock;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
 
 @ExtendWith(ApplicationExtension.class)
 public class MinesweeperControllerTest {
     private MinesweeperController controller;
     private final int boardSize = 20;
     private MinesweeperCell[][] board = new MinesweeperCell[boardSize][boardSize];
     private GridPane gridPane;
-    // how to test constructor? especially since constructor calls placeMines() method
-    @Test
-    public void initialize(){
+
+    @Test //passes, incomplete
+    public void constructController(){
+        //given
+        givenMinesweeperController();
+
+        //when
+
+        //then
+        assertNotNull(controller.getBoard());
+        //assert placeMines method was called
```

Comment:
You just need to assert that there are mine and that the numbers in the cells are correct.